src/main/kotlin/anttracker/Request.kt

```diff
@@ -12,92 +12,353 @@ package anttracker.request
+        when (val selection = readln()) {
+            // abort early - don't select an issue
+            "`" -> return null
+
+            // go to next page
+            "" -> {
+                if (!issuePage.lastPage()) {
+                    issuePage.loadNextPage()
+                } else {
+                    println("No more pages to display.")
+                }
+            }
+
+            // else: check for line # selection
+            else -> {
+                try {
+                    // will throw java.lang.NumberFormatException if not int
+                    val userInputInt = selection.toInt()
+
+                    // validate integer selection
+                    if (issuePage.isValidLineNum(userInputInt)) {
+                        linenum = userInputInt
+                    }
+                } catch (e: java.lang.NumberFormatException) {
+                    // print error message, then re-prompt user
+                    println(e.message)
+                    // continue
+                }
+            }
+        }
+    }
+
+    // return selected issue record
+    return issuePage.getContentAt(linenum - 1)
+}
+
+// ----------------------------------------------------------------
+
+/* This function collects from the user all the information needed to create an issue
+ * by prompting them for the description, affectedRelease, and priority.
+ * Returns null if the user indicates to abort the procedure.
+ * The created issue has creation date set to the time of insertion, and status of created. */
+private fun enterIssueInformation(
+    product: ProductEntity, // in
+): Issue? {
+    var desc: String? = null
+
+    // select description of issue; loops until a valid desc is entered
+    while (desc == null) {
+        // prompt user for input
+        println("Issue description (max 30 chars). ` to abort:")
+
+        when (val inputDesc = readln()) {
+            // user aborts
+            "`" -> return null
+
+            // user input a string
+            else -> {
+                if (inputDesc.isEmpty() || inputDesc.length > 30) {
+                    println("Bad input.")
+                    // continue
+                } else {
+                    desc = inputDesc
+                    // loop ends here
+                }
+            }
+        }
+    }
+
+    // select anticipated release for this issue
+    val release = selectRelease(product.name) ?: return null
+
+    var priority: Short? = null
+
+    // select the priority of this issue; loops until a valid priority is chosen
+    while (priority == null) {
+        print("Issue priority (1-5). ` to abort: ")
+
+        when (val selection = readln()) {
+            // user aborts
+            "`" -> return null
+
+            // user input needs validation
+            else -> {
+                try {
+                    // will throw java.lang.NumberFormatException if not int
+                    val userInputShort = selection.toShort()
```

Comment:
It would be a better user experience to only accept valid inputs. You could create a function which given a prompt and a set of choices, only returns the user's input if it is among that set of choices. 

I made something similar in the `Terminal` class called `prompt`.
https://github.com/AntTracker/AntTracker/blob/34c055a8b54cdce21c46cf6788f9d2bbe3cb359a/src/main/kotlin/anttracker/issues/Terminal.kt#L44

Reply:
I get where you're coming from, but this module is already quite large. I'd prefer to keep things as is, and the validation being done is simple as is

Reply:
I do not think calling a function from another module will significantly increase the size of the request module, but I will not force you.